### PR TITLE
Configure jest testing

### PR DESCRIPTION
**Pull request by smcclure17**
I plan to add actual tests in follow-up PRs, but I'm adding the setup here in hopes not to bloat them

### CODE REVIEW

File: functions/src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe("Hello World", () => {
+    test("Placeholder Test", () => {
```
Comment (mikelehen) [inline functions/src/index.test.ts]: Surprised to see 4 space indents instead of 2.  I guess we don't have prettier set up to automatically run?

**Review by mikelehen — APPROVED** [review]
Looks good!

**Review by mikelehen — APPROVED** [review]
Looks good!

File: functions/src/index.test.ts (hunk 1 above)
Comment (smcclure17) [reply functions/src/index.test.ts]: Ah, we haven't set up a prettier pre-commit hook, so the file isn't getting formatted. I've now run it manually in 8e3e63e695a84ed5cc8bda407dae759632c3a09e but I'll add a ticket to add the hook

**Pull request merged**